runtime/interp.c

```diff
@@ -1326,7 +1326,9 @@ do_resume: {
 
       check_trap_barrier_for_effect (domain_state);
       if (parent_stack == NULL) {
+        Setup_for_gc;
         accu = caml_make_unhandled_effect_exn(accu);
+        Restore_after_gc;
```

Comment:
This should be `Setup_for_c_call`. `Setup_for_gc` is incorrect here as it will save and restore the `accu` parameter, un-doing the work of line 1330.

(Is this true? If it happens as I described above it should have been caught by testing.)

Reply:
Hmm, not sure how that's got through - the first version was `Setup_for_c_call` and I then convinced myself that was the wrong the pairing. I'll have a look at this again tomorrow.

Reply:
Fixed to use `Setup_for_c_call` - the difference with the `pc` storage here doesn't matter